dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.ChatCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.AzureSdk;
+
+public abstract class ClientBase
+{
+    protected string ModelId { get; set; } = string.Empty;
+    protected OpenAIClient? Client { get; set; }
+
+    /// <summary>
+    /// Creates a completion for the prompt and settings.
+    /// </summary>
+    /// <param name="text">The prompt to complete.</param>
+    /// <param name="requestSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Text generated by the remote model</returns>
+    protected async Task<string> InternalCompleteTextAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(requestSettings, "Completion settings cannot be empty");
+
+        if (requestSettings.MaxTokens < 1)
+        {
+            throw new AIException(
+                AIException.ErrorCodes.InvalidRequest,
+                $"MaxTokens {requestSettings.MaxTokens} is not valid, the value must be greater than zero");
+        }
+
+        var options = new CompletionsOptions
+        {
+            Prompts = { text.NormalizeLineEndings() },
+            MaxTokens = requestSettings.MaxTokens,
+            Temperature = (float?)requestSettings.Temperature,
+            NucleusSamplingFactor = (float?)requestSettings.TopP,
+            FrequencyPenalty = (float?)requestSettings.FrequencyPenalty,
+            PresencePenalty = (float?)requestSettings.PresencePenalty,
+            Echo = false,
+            ChoicesPerPrompt = 1,
+            GenerationSampleCount = 1,
+            LogProbabilityCount = null,
+            User = null,
+        };
+
+        if (requestSettings.StopSequences is { Count: > 0 })
+        {
+            foreach (var s in requestSettings.StopSequences)
+            {
+                options.StopSequences.Add(s);
+            }
+        }
+
+        Response<Completions>? response = null;
+        try
+        {
+            response = await this.Client!.GetCompletionsAsync(
+                this.ModelId,
+                options,
+                cancellationToken);
+        }
+        catch (RequestFailedException e)
+        {
+            switch (e.Status)
+            {
+                case (int)HttpStatusCodeType.BadRequest:
```

Comment:
The same logic for mapping response status code to AIException lives in OpenAIClientBase class. Consider putting it in one place.

Reply:
the logic is a bit different because the other one uses enums, this uses int. Open to suggestions but I'd rather do this later. The tech debt is isolated inside this lib, so it's manageable.